components/centraldashboard/public/components/notebook-default-card.js

```diff
@@ -0,0 +1,129 @@
+import '@polymer/iron-ajax/iron-ajax.js';
+import '@polymer/iron-icon/iron-icon.js';
+import '@polymer/iron-icons/iron-icons.js';
+import '@polymer/paper-card/paper-card.js';
+import '@polymer/iron-icons/social-icons.js';
+import '@polymer/paper-toast/paper-toast.js';
+import '@polymer/paper-ripple/paper-ripple.js';
+import '@polymer/paper-item/paper-icon-item.js';
+import '@polymer/paper-icon-button/paper-icon-button.js';
+
+import {html, PolymerElement} from '@polymer/polymer';
+
+import './card-styles.js';
+import css from './notebook-default-card.css';
+import template from './notebook-default-card.pug';
+// eslint-disable-next-line max-len
+import {AppLocalizeBehavior} from '@polymer/app-localize-behavior/app-localize-behavior.js';
+import {mixinBehaviors} from '@polymer/polymer/lib/legacy/class.js';
+
+// eslint-disable-next-line max-len
+export class NotebookDefaultCard
+    extends mixinBehaviors([AppLocalizeBehavior], PolymerElement) {
+    static get template() {
+        return html([`
+        <style include="card-styles">
+            ${css.toString()}
+        </style>
+            ${template()}
+        `]);
+    }
+
+    /**
+     * Object describing property-related metadata used by Polymer features
+     */
+    static get properties() {
+        return {
+            defaultNotebookError: String,
+            namespace: {
+                type: String,
+                observer: '_namespaceChanged',
+                value: null,
+            },
+            loading: {type: Boolean, value: false},
+        };
+    }
+
+    /**
+     * Retrieves default notebook when namespace is selected.
+     * @param {string} newNamespace
+     */
+    _namespaceChanged(newNamespace) {
+        if (!newNamespace) return;
+        this.$['GetDefaultNotebook'].generateRequest();
+    }
+
+    _isNotebookReady() {
+        return this.defaultNotebook.notebook.status.phase == 'ready';
+    }
+
+    _getNamespace() {
```

Comment:
Is this function relevant? I don't see it called anywhere and its returning either a boolean or a string, which is confusing.

Reply:
Deleted